enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,609 @@
+This process can be done using the ListBundles gRPC registry API, which will return a stream of bundles that are present in that catalog's index image.
+
+An initial implementation for the "best bundle" selection criteria may include selecting the highest semver bundle that's within the cluster version range requirements. Determining the cluster version information can be done by using a combination of the Kubernetes' discovery API, and the ClusterVersion resource.
+Future phases may aim to improve this bundle selection implementation through constructing upgrade graphs, evaluating package/cluster constraints, and performing dependency resolution.
+
+Once the list of available bundles within a catalog has been returned, additional filtering logic may be performed. As a result, a single registry+v1 bundle will be returned, and that bundle's metadata will be passed to the PO manager's installation layer for further processing.
+
+#### Applying Bundles
+
+```mermaid
+sequenceDiagram
+participant PO Manager
+participant PO Installer
+participant BundleDeployment
+participant plain provisioner
+participant Bundle
+participant registry provisioner
+PO Manager->>PO Installer: Desired registry+v1 bundle
+PO Installer->>BundleDeployment: Generates
+plain provisioner->>Bundle: Generates
+plain provisioner->>BundleDeployment: Manages
+registry provisioner->>Bundle: Unpacks
+registry provisioner-->Bundle: Converts OLM's CSV resource into RBAC/Deployment/etc.
+registry provisioner-->Bundle: Generates health probes for resources
+registry provisioner->>Bundle: Stores converted bundle contents
+plain provisioner->>Bundle: Waits for Bundle to unpack
+plain provisioner->>Bundle: Loads bundle contents
+plain provisioner-->Bundle: Applies loaded bundle contents
+```
+
+After a single resolved registry+v1 bundle has been selected by the PO manager's sourcing layer, a BundleDeployment resource will be generated that will manage those bundle contents. This generated resource will contain an owner reference that points back to the PO custom resource the manager is currently reconciling.
+
+> Note: The PO API is a cluster-scoped resource, and therefore placing owner references on an underlying, cluster-scoped BundleDeployment resource is a supported workflow in Kubernetes.
+
+When generating the BundleDeployment resource, the PO manager will use the unique "core.rukpak.io/plain" provisioner ID to manage and apply the desired Bundle template, and the "core.rukpak.io/registry" provisioner ID to source the desired registry+v1 bundle contents.
+
+An example of this behavior can be seen below:
+
+```yaml=
+apiVersion: core.rukpak.io/v1alpha1
+kind: BundleDeployment
+metadata:
```

Comment:
Should the PO controller inject its own labels into BundleDeployments so that PO-related controllers can optimize listing/watching? Otherwise, we'd list/watch _all_ BDs and then filter client-side on owner ref.

Reply:
I think my immediate thoughts are definitely yes, but it could also be potentially premature until we have a clearer picture of rukpak consumers? Maybe this is something we can revisit after the initial design, unless you had ideas on how these labels could be consumed.

Reply:
Added this to the answered questions in https://github.com/openshift/enhancements/pull/1170/commits/31269f34570363358512ef7e14a4da757e779476.